inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp

```diff
@@ -203,18 +206,48 @@ void Engine::AddExtension(InferenceEngine::IExtensionPtr extension) {
 }
 
 void Engine::QueryNetwork(const ICNNNetwork& network, const std::map<std::string, std::string>& config, QueryNetworkResult& res) const {
-    details::CNNNetworkIterator i(&network);
-    while (i != details::CNNNetworkIterator()) {
-        try {
-            mkldnn::engine eng(mkldnn::engine(mkldnn::engine::kind::cpu, 0));
-            MKLDNNWeightsSharing::Ptr fake_w_cache;
-
-            // if we can create and have not thrown exception, then layer is supported
-            std::unique_ptr <MKLDNNNode>(MKLDNNNode::CreateNode(*i, eng, extensionManager, fake_w_cache));
-            res.supportedLayersMap.insert({ (*i)->name, GetName() });
-        } catch (InferenceEngine::details::InferenceEngineException&) {
+    MKLDNNWeightsSharing::Ptr fake_w_cache;
+    auto function = network.getFunction();
+    if (function != nullptr) {
+        std::unordered_set<std::string> originalOps;
+        {
+            auto clonedNetwork = cloneNetwork(network);
+            for (details::CNNNetworkIterator itLayer{clonedNetwork.get()}; itLayer != details::CNNNetworkIterator(); itLayer++) {
```

Comment:
`details::CNNNetworkIterator` leads to conversion to old representation.. `itLayer->name` is `CNNLayer`, while you should work with `ngraph::Op`

Reply:
Used just to get names from **cloned** network.

Reply:
removed